api/v1/invitations/InvitationController.php

```diff
@@ -124,7 +126,7 @@ public function getGroupRoutes(): void
             ]),
         ])->group(function () {
 
-            Route::get('', $this->getMany(...))
+            Route::get('{type}', $this->getMany(...))
```

Comment:
@defstat this change is causing all tests break with latest pull of `pkp-lib` `main` branch . With the add of route param `type`, which probably has not been considered in some other places(from where AJAX request are made) cause the following exception
```
[29-Dec-2024 08:20:56 UTC] Symfony\Component\HttpKernel\Exception\NotFoundHttpException: The route publicknowledge/api/v1/invitations could not be found. in /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:44
Stack trace:
#0 /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Routing/RouteCollection.php(163): Illuminate\Routing\AbstractRouteCollection->handleMatchedRoute(Object(Illuminate\Http\Request), NULL)
#1 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/core/PKPBaseController.php(104): Illuminate\Routing\RouteCollection->match(Object(Illuminate\Http\Request))
#2 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/middleware/traits/HasRequiredMiddleware.php(53): PKP\core\PKPBaseController::getRequestedRoute(Object(Illuminate\Http\Request))
#3 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/middleware/DecodeApiTokenWithValidation.php(64): PKP\middleware\DecodeApiTokenWithValidation->hasRequiredMiddleware(Object(Illuminate\Http\Request))
#4 /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PKP\middleware\DecodeApiTokenWithValidation->handle(Object(Illuminate\Http\Request), Object(Closure))
#5 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/middleware/SetupContextBasedOnRequestUrl.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#6 /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PKP\middleware\SetupContextBasedOnRequestUrl->handle(Object(Illuminate\Http\Request), Object(Closure))
#7 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/middleware/AllowCrossOrigin.php(34): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#8 /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PKP\middleware\AllowCrossOrigin->handle(Object(Illuminate\Http\Request), Object(Closure))
#9 /Users/abir/Sites/code/ojs-main/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#10 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/handler/APIHandler.php(102): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#11 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/core/APIRouter.php(116): PKP\handler\APIHandler->runRoutes()
#12 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/core/Dispatcher.php(158): PKP\core\APIRouter->route(Object(APP\core\Request))
#13 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/core/PKPApplication.php(395): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#14 /Users/abir/Sites/code/ojs-main/index.php(30): PKP\core\PKPApplication->execute()
#15 /Users/abir/.composer/vendor/laravel/valet/server.php(110): require('/Users/abir/Sit...')
#16 {main}
```
As the route has changed with the addition of a new route param, route is not found by router .

Reply:
@touhidurabir, sorry about that - I have made the change required and waiting for the tests to finish.

Reply:
@defstat in the error log of failed tests , I can see the following exception 
```
[Mon Dec 30 12:03:05 2024] Symfony\Component\HttpKernel\Exception\NotFoundHttpException: The route publicknowledge/api/v1/invitations/userRoleAssignment could not be found. in /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:44
```
but I can tests with latest `pkp-lib` `main` checkout, this works for me . 

However the current `pkp-lib` submodule reference to main app is not the latest main but few commits behind where the route in `InvitationController` set as 
```php
Route::get('', $this->getMany(...))
                ->name('invitation.getMany');
```
I think thats causing the issue . I think you need include the latest `pkp-lib` submodule update also in your PR .

Reply:
Thanks @touhidurabir! again it seems that the tests are failing down the PHPUnit part of the tests this time.

Reply:
Tests are failing because of this 
```
Test Errored (functional\ArticleBodyTest::testCreate)
array_reduce(): Argument #1 ($array) must be of type array, null given
After Test Method Called (functional\ArticleBodyTest::tearDown)
```

it's a issue with `jatsTemplate` plugin's unit test and I have added a fix in the PR of pkp/pkp-lib#10292 at https://github.com/pkp/jatsTemplate/pull/54/files#diff-c87e6397a6dd5c2bf754d5b10da491be897899cf411d80e1dbcefedb20bd9960R114